api/v1beta1/common_types.go

```diff
@@ -122,6 +122,12 @@ const (
 	PKIBackendK8sCSR PKIBackend = "k8s-csr"
 )
 
+// ObjectReference is a reference to a kubernetes object
+type ObjectReference struct {
```

Comment:
The name `ObjectReference` might be confusing as it's used by Kubernetes as well. I'd name it something like `IstioControlPlaneReference`